diff --git a/m_menu.c b/m_menu.c
index c0b3eb0..6ab4fc0 100644
--- a/m_menu.c
+++ b/m_menu.c
@@ -377,9 +377,6 @@ menu_t  NewDef =
 enum
 {
     endgame,
-    detail,
-    scrnsize,
-    option_empty1,
     mousesens,
     option_empty2,
     soundvol,
@@ -389,9 +386,6 @@ enum
 menuitem_t OptionsMenu[]=
 {
     {1,"M_ENDGAM",	M_EndGame,'e'},
-    {1,"M_DETAIL",	M_ChangeDetail,'g'},
-    {2,"M_SCRNSZ",	M_SizeDisplay,'s'},
-    {-1,"",0},
     {2,"M_MSENS",	M_ChangeSensitivity,'m'},
     {-1,"",0},
     {1,"M_SVOL",	M_Sound,'s'}
@@ -1070,15 +1064,9 @@ char    detailNames[2][9]	= {"M_GDHIGH","M_GDLOW"};
 void M_DrawOptions(void)
 {
     V_DrawPatchDirect (108,15,0,W_CacheLumpName("M_OPTTTL",PU_CACHE));
-	
-    V_DrawPatchDirect (OptionsDef.x + 175,OptionsDef.y+LINEHEIGHT*detail,0,
-		       W_CacheLumpName(detailNames[detailLevel],PU_CACHE));
 
     M_DrawThermo(OptionsDef.x,OptionsDef.y+LINEHEIGHT*(mousesens+1),
 		 10,mouseSensitivity);
-	
-    M_DrawThermo(OptionsDef.x,OptionsDef.y+LINEHEIGHT*(scrnsize+1),
-		 9,screenSize);
 }
 
 void M_Options(int choice)
